subprojects/model-core/src/main/java/org/gradle/internal/credentials/DefaultPasswordCredentials.java

```diff
@@ -39,6 +42,7 @@ public void setUsername(String username) {
         this.username = username;
     }
 
+    @Internal
```

Comment:
Won't this mean that if the password changes, it won't force the task to be marked as out-of-date?

Reply:
It does, and that is fine. Passwords do not change the outputs of the task, only if it succeeds or not. So if you already have a successful execution with one password, why would you need to run the same thing again with a different password?